fl4health/strategies/feddg_ga_strategy.py

```diff
@@ -0,0 +1,411 @@
+from enum import Enum
+from logging import WARNING
+from typing import Callable, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+from flwr.common import MetricsAggregationFn, NDArrays, Parameters, ndarrays_to_parameters, parameters_to_ndarrays
+from flwr.common.logger import log
+from flwr.common.typing import EvaluateRes, FitIns, FitRes, Scalar
+from flwr.server.client_manager import ClientManager
+from flwr.server.client_proxy import ClientProxy
+from flwr.server.strategy import FedAvg
+
+from fl4health.client_managers.fixed_sampling_client_manager import FixedSamplingClientManager
+
+
+class FairnessMetricType(Enum):
+    """Defines the basic types for fairness metrics, their default names and their default signals"""
+
+    ACCURACY = "val - prediction - accuracy"
+    LOSS = "val - loss"
+    CUSTOM = "custom"
+
+    @classmethod
+    def signal_for_type(cls, fairness_metric_type: "FairnessMetricType") -> float:
+        """
+        Return the default signal for the given metric type.
+
+        Args:
+            fairness_metric_type: (FairnessMetricType) the fairness metric type.
+
+        Returns: (float) -1.0 if FairnessMetricType.ACCURACY or 1.0 if FairnessMetricType.LOSS.
+
+        Raises: (Exception) if type is CUSTOM as the signal has to be defined by the user.
+        """
+        # For loss values, large and **positive** gaps imply worse generalization of global
+        # weights to local models. Therefore, we want to **increase** weight for these model
+        # parameters to improve generalization. So signal is positive. For accuracy, large
+        # **negative** gaps imply worse generalization. So the signal is -1.0, to increase
+        # weights for the associated model parameters.
+        if fairness_metric_type == FairnessMetricType.ACCURACY:
+            return -1.0
+        if fairness_metric_type == FairnessMetricType.LOSS:
+            return 1.0
+        raise Exception("This function should not be called with CUSTOM type.")
```

Comment:
Super minor, but perhaps we use a specific exception class rather than a bare one?